### PR TITLE
Perform type assertion on conversion

### LINKED ISSUE #2926: kafka-controller panic: interface conversion: interface {} is cache.DeletedFinalStateUnknown, not *v1alpha1.Consumer
Observed a panic in kafka-controller

```
E0126 08:17:37.130870       1 runtime.go:79] Observed a panic: &runtime.TypeAssertionError{_interface:(*runtime._type)(0x1d15700), concrete:(*runtime._type)(0x1e458a0), asserted:(*runtime._type)(0x20493c0), missingMethod:""} (interface conversion: interface {} is cache.DeletedFinalStateUnknown, not *v1alpha1.Consumer)
goroutine 650 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x1d96260?, 0xc00a7161e0})
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc007f15f18?})
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:49 +0x75
panic({0x1d96260, 0xc00a7161e0})
	/usr/lib/golang/src/runtime/panic.go:838 +0x207
knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/consumergroup.enqueueConsumerGroupFromConsumer.func1({0x1e458a0?, 0xc00a46aa60?})
	/remote-source/knative-kafka-broker/app/control-plane/pkg/reconciler/consumergroup/controller.go:168 +0x85
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnDelete(...)
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/client-go/tools/cache/controller.go:246
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/client-go/tools/cache/shared_informer.go:820 +0xef
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x11?)
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:157 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000db6f38?, {0x240d1e0, 0xc0006424e0}, 0x1, 0xc0009b0000)
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:158 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0?, 0x3b9aca00, 0x0, 0x0?, 0x0?)
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:135 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:92
k8s.io/client-go/tools/cache.(*processorListener).run(0xc000166e80?)
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/client-go/tools/cache/shared_informer.go:812 +0x6b
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:75 +0x5a
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:73 +0x85
panic: interface conversion: interface {} is cache.DeletedFinalStateUnknown, not *v1alpha1.Consumer [recovered]
	panic: interface conversion: interface {} is cache.DeletedFinalStateUnknown, not *v1alpha1.Consumer

goroutine 650 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc007f15f18?})
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:56 +0xd8
panic({0x1d96260, 0xc00a7161e0})
	/usr/lib/golang/src/runtime/panic.go:838 +0x207
knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/consumergroup.enqueueConsumerGroupFromConsumer.func1({0x1e458a0?, 0xc00a46aa60?})
	/remote-source/knative-kafka-broker/app/control-plane/pkg/reconciler/consumergroup/controller.go:168 +0x85
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnDelete(...)
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/client-go/tools/cache/controller.go:246
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/client-go/tools/cache/shared_informer.go:820 +0xef
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x11?)
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:157 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000db6f38?, {0x240d1e0, 0xc0006424e0}, 0x1, 0xc0009b0000)
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:158 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0?, 0x3b9aca00, 0x0, 0x0?, 0x0?)
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:135 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:92
k8s.io/client-go/tools/cache.(*processorListener).run(0xc000166e80?)
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/client-go/tools/cache/shared_informer.go:812 +0x6b
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1()
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:75 +0x5a
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
	/remote-source/knative-kafka-broker/app/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:73 +0x85
```

The container restart count was 1, and the panic was at the end of the logs of the "--previous" instance, so assuming it  caused a crash.

### PR DESCRIPTION
**Pull request by matzew**
Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

Fixes #2926

## Proposed Changes

- as per title

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: control-plane/pkg/reconciler/consumergroup/controller.go [hunk 1]
```diff
@@ -167,10 +167,14 @@ func ResyncOnStatefulSetChange(ctx context.Context, handle func(interface{})) {
...
-			enqueue(types.NamespacedName{Namespace: c.GetNamespace(), Name: or.Name})
+		c, ok := obj.(*kafkainternals.ConsumerGroup)
+		if ok {
+			or := c.GetVReplicas()
+			if or != 0 {
+				enqueue(types.NamespacedName{Namespace: c.GetNamespace(), Name: "F"})
+			}
+		} else {
+			time.Sleep(50* time.Second)
 		}
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/consumergroup/controller.go:178]: why sleep?

Comment (matzew) [reply control-plane/pkg/reconciler/consumergroup/controller.go]: removed - was for local debugs :man_facepalming:

**Review by pierDipi — APPROVED** [review]
/lgtm

**Pull request merged**